src/Renderer/NavigatorRenderer.cpp

```diff
@@ -0,0 +1,533 @@
+      w = Layout::Scale(2);
+
+    } else if (i < summary.active) {
+      if (it->achieved)
+        canvas.Select(look_task.hbGreen);
+      else
+        canvas.Select(look_task.hbNotReachableTerrain);
+      w = Layout::Scale(2);
+
+    } else {
+      if (it->achieved)
+        canvas.Select(look_task.hbGreen);
+      else
+        canvas.Select(look_task.hbLightGray);
+
+      w = Layout::Scale(1);
+    }
+
+    canvas.DrawRectangle(PixelRect{position_waypoint}.WithMargin(w));
+
+    }
+  }
+
+  void 
+  NavigatorRenderer::DrawWaypointsIconsTitle(Canvas &canvas, const NavigatorLook &look, bool inverse) noexcept{
+    
+    const int rc_height = canvas.GetHeight();
+    const int rc_width = canvas.GetWidth();
+
+    const WaypointRendererSettings &waypoint_settings = CommonInterface::GetMapSettings().waypoint;
+    const WaypointLook &waypoint_look = UIGlobals::GetMapLook().waypoint;
+    
+    WaypointIconRenderer waypoint_icon_renderer{waypoint_settings, waypoint_look, canvas};
+    const PixelPoint position_waypoint_left{rc_width*7/200, rc_height*1/2};
+    // const PixelPoint position_waypoint_centered{rc_width*42/200, rc_height*1/2};
+    const PixelPoint position_waypoint_right{rc_width*190/200, rc_height*1/3};
+    
+    WaypointPtr waypoint_before;
+    WaypointPtr waypoint_current;
+
+    unsigned task_size{};
+    unsigned i{};
+    
+    if(protected_task_manager != nullptr) {
+      ProtectedTaskManager::Lease lease(*protected_task_manager);
+
+      const OrderedTask &task = lease->GetOrderedTask();
+
+      task_size = task.TaskSize();
+      waypoint_current = task.GetActiveTaskPoint()->GetWaypointPtr();
+      i = task.GetActiveIndex();
+      
+      if(i == 0)
+        waypoint_before = task.GetPoint(0).GetWaypointPtr();
+      else
+        waypoint_before = task.GetPoint(i-1).GetWaypointPtr();
+
+
+      // std::cout << "\nhas started? " << task.TaskStarted() << "\nhas finished? " << task.GetStats().task_finished << std::endl;
+    }
+    
+    // CALCULATE REACHABILITY
+    WaypointReachability wr_before{};
+    WaypointReachability wr_current{};
+
+    const auto &basic = CommonInterface::Basic();
+    const ComputerSettings &settings_computer = CommonInterface::GetComputerSettings();
+    const TaskBehaviour &task_behaviour = settings_computer.task;
+    const auto &calculated = CommonInterface::Calculated();
+    const GlidePolar &glide_polar = CommonInterface::GetComputerSettings().polar.glide_polar_task;
+    const MacCready mac_cready(task_behaviour.glide, glide_polar);
+
+    for (auto wp : {waypoint_before, waypoint_current}) {
+      const auto elevation_wp_before = wp->elevation + task_behaviour.safety_height_arrival;
+      const GlideState state_wp_before{GeoVector(basic.location, wp->location), 
+                            elevation_wp_before, basic.nav_altitude, calculated.GetWindOrZero()};
+      const GlideResult result_wp_before = mac_cready.SolveStraight(state_wp_before);
```

Comment:
Don't do expensive calculations in a renderer!

Reply:
For the moment, I let this functionality aside the project.
I commented and added it in the TODO list.